autogpt_platform/backend/backend/data/execution.py

```diff
@@ -21,6 +21,35 @@
 from backend.util.settings import Config
 
 
+class ExecutionStats(BaseModel):
+    """Execution statistics for a graph execution."""
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    error: Optional[Exception] = None
+    walltime: float = 0
+    cputime: float = 0
+    nodes_walltime: float = 0
+    nodes_cputime: float = 0
+    node_count: int = 0
+    node_error_count: int = 0
+    cost: float = 0
+
+
+class ExecutionStatsStrErr(BaseModel):
```

Comment:
Why can't I extend the above class and override the error field

Reply:
Also, do we really need two versions of this? Can we just stick to this class only ?

Reply:
Exception isn't serializable and pydantic will complain about it, figured this was the easiest way without building and registering searlizers while still keeping the data I need for it for handling the errors

Reply:
```python
/Users/ntindle/code/agpt/AutoGPT/autogpt_platform/backend/backend/data/execution.py
  /Users/ntindle/code/agpt/AutoGPT/autogpt_platform/backend/backend/data/execution.py:43:5 - error: "error" overrides symbol of same name in class "ExecutionStats"
    Variable is mutable so its type is invariant
      Override type "str | None" is not the same as base type "Exception | None" (reportIncompatibleVariableOverride)
    ```
    
    We have a rule against it, and pydantic will serialize it given the opportunity so we'd be doing fucky stuff with `del attr`

Reply:
can do optional[whatever] but then you just end up with a bunch of null fields in the db

Reply:
IMO, having one none field seems fine compared to complicating the code with more class duplications.

TBF you have introduced more null fields in the db by introducing Pydantic class for the dynamic dict stats, now we are forcing all blocks persisting `llm_token_count` and other non-specific metrics with null/zero values for node execution.